### PR TITLE
[GEN-276] Add option to display disabled perils

### PR DESCRIPTION
**Pull request by robinandeer**
## Describe your changes

- Show missing persils for a tier as disabled

- Add option to turn this off on the block

## Justify why they are needed

Request from Peter

## Checklist before requesting a review

- [ ] I have performed a self-review of my code

### CODE REVIEW

**Comment by robinandeer** [comment]
Current dependencies on/for this PR:
* main
  * **PR #2563** <a href="https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2563" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/HedvigInsurance/racoon/2563?utm_source=stack-comment).

File: apps/store/src/blocks/PerilsBlock.tsx [hunk 1]
```diff
@@ -8,11 +8,25 @@ import { SbBaseBlockProps } from '@/services/storyblok/storyblok'
...
   heading?: string
+  hideDisabledPerils?: boolean
 }>
 
 export const PerilsBlock = ({ blok }: PerilsBlockProps) => {
   const { productData, selectedVariant } = useProductPageContext()
 
+  const allPerils = useMemo(() => {
+    const addedPerils = new Set<string>()
+    return productData.variants
```
Comment (alebedev) [inline apps/store/src/blocks/PerilsBlock.tsx:19]: Do we need side effects here?  If only thing we're doing is making values unique, try using Set constructor
```
   new Set(productData.variants.flatMap(item => item.perils))
```

**Review by alebedev — APPROVED** [review]
Looks good, one comment

File: apps/store/src/blocks/PerilsBlock.tsx (hunk 1 above)
Comment (robinandeer) [reply apps/store/src/blocks/PerilsBlock.tsx:19]: Hmm but objects compare by identity so we the set will contain peril duplicates if we only do that 🤔

**Pull request merged**